src/error-catch-notifier.js

```diff
@@ -0,0 +1,235 @@
+/**
+ * Error catching is enabled
+ * @member {bool}
+ */
+let _isEnabled = false;
+
+/**
+ * Error logging is enabled
+ * @member {bool}
+ */
+let _isLoggingEnabled = false;
+
+/**
+ * A function that takes as its first argument a parameter named error
+ * The first argument of an errorSubscriber must be named error
+ * @typedef {function} ErrorSubscriber
+ * @param {error} error - error object
+ */
+
+/**
+ * Array of functions called when an error is caught
+ * @member {function[]} errorSubscriberFunctions
+ * @param {ErrorSubscriber} errorSubscriberFunctions[]
+ */
+let _errorSubscribers = [];
+
+/**
+ * Returns an array of argument names from a function
+ * @param {function} targetFunction - function definition
+ * @return {string[]} names of function definition's arguments
+ */
+export function getArgumentNames(targetFunction) {
+    // First match everything inside the function argument parens.
+    let args;
+    try {
+        args = targetFunction.toString().match(/function\s.*?\(([^)]*)\)/)[1];
+    } catch (error) {
+        return [];
+    }
+
+    // Split the arguments string into an array comma delimited.
+    return args.split(',').map(function argsTrimCommentsAndWhiteSpace(arg) {
+    // Ensure no inline comments are parsed and trim the whitespace.
+        return arg.replace(/\/\*.*\*\//, '').trim();
+    }).filter(function filterUndefined(arg) {
+    // Ensure no undefined values are added.
+        return arg;
+    });
+}
+
+/**
+ * Determines if object is a function
+ * @param {object} obj - any value with object on its prototype chain
+ * @return {bool} true if obj is a function; false if not
+ */
+export function isFunction(obj) {
+    return !!(obj && obj.constructor && obj.call && obj.apply);
```

Comment:
@mscienski I thought about using a module but decided I'd just copy pastaed an implementation so the module could be dependency free.